File: importer/src/main/java/org/hiero/mirror/importer/repository/AddressBookEntryRepository.java
// SPDX-License-Identifier: Apache-2.0

package org.hiero.mirror.importer.repository;

import org.hiero.mirror.common.domain.addressbook.AddressBookEntry;
import org.springframework.data.repository.CrudRepository;

public interface AddressBookEntryRepository extends CrudRepository<AddressBookEntry, AddressBookEntry.Id> {}

File: rosetta/app/interfaces/address_book_entry_repository.go
// SPDX-License-Identifier: Apache-2.0

package interfaces

import (
	"context"

	rTypes "github.com/coinbase/rosetta-sdk-go/types"
	"github.com/hiero-ledger/hiero-mirror-node/rosetta/app/domain/types"
)

// AddressBookEntryRepository Interface that all AddressBookEntryRepository structs must implement
type AddressBookEntryRepository interface {

	// Entries return all current address book Entries
	Entries(ctx context.Context) (*types.AddressBookEntries, *rTypes.Error)
}
